pkg/kn/commands/types.go

```diff
@@ -31,12 +31,17 @@ var KubeCfgFile string
 type KnParams struct {
 	Output         io.Writer
 	ServingFactory func() (serving.ServingV1alpha1Interface, error)
+	// CurrentNamespace is the namespace set in the current context.
+	CurrentNamespace string
 }
 
-func (c *KnParams) Initialize() {
+func (c *KnParams) Initialize() error {
 	if c.ServingFactory == nil {
 		c.ServingFactory = GetConfig
 	}
+	var err error
+	c.CurrentNamespace, err = GetCurrentNamespace(KubeCfgFile)
```

Comment:
I wonder whether we could get rid of sharing that global variable and use a function instead which could cache a lookup. This would free us for knowing the order when this variable was initialized (which I just checked, looks good, but you don't know what happens when people refactor code).

I.e. introducing a `GetKubeCfgFile()` function which checks a private variable kubeCfgFile and, if empty, does the actual lookup as it's done in `InitializeConfig`:


```go
func GetKubeCfgFile() string {
   if kubeCfgFile == "" {
       kubeCfgFile = initKubeConfig()
   }
   return kubeCfgFile
}
```

(and maybe improve the error handling over there, as initKubeConfig() must not call `os.Exit()` but should return an error which should be propagated here, too.

Reply:
Agree, I made it public and hated that. Better to go back to old Java-style getters for this one and hide the var. Thanks for suggesting this.